xcdat/temporal.py

```diff
@@ -816,75 +904,52 @@ def _drop_incomplete_djf(self, dataset: xr.Dataset) -> xr.Dataset:
+
         for year_month in incomplete_seasons:
             try:
                 coord_pt = ds.loc[dict(time=year_month)][self._dim][0]
                 ds_time = ds_time.where(ds_time[self._dim] != coord_pt, drop=True)
-                self._time_bounds = ds_time[self._time_bounds.name]
             except (KeyError, IndexError):
                 continue
 
         ds_final = xr.merge((ds_time, ds_no_time))
 
         return ds_final
 
-    def _form_seasons(self, custom_seasons: List[List[str]]) -> Dict[str, List[str]]:
-        """Forms custom seasons from a nested list of months.
+    def _drop_leap_days(self, ds: xr.Dataset):
+        """Drop leap days from time coordinates.
 
-        This method concatenates the strings in each sublist to form a
-        a flat list of custom season strings
+        This method is used to drop 2/29 from leap years (if present) before
+        calculating climatology/departures for high frequency time series data
+        to avoid `cftime` breaking (`ValueError: invalid day number provided
+        in cftime.DatetimeProlepticGregorian(1, 2, 29, 0, 0, 0, 0,
+        has_year_zero=True`).
 
         Parameters
         ----------
-        custom_seasons : List[List[str]]
-            List of sublists containing month strings, with each sublist
-            representing a custom season.
+        ds : xr.Dataset
+            The dataset.
 
         Returns
         -------
-        Dict[str, List[str]]
-           A dictionary with the keys being the custom season and the
-           values being the corresponding list of months.
-
-        Raises
-        ------
-        ValueError
-            If exactly 12 months are not passed in the list of custom seasons.
-        ValueError
-            If a duplicate month(s) were found in the list of custom seasons.
-        ValueError
-            If a month string(s) is not supported.
+        xr.Dataset
         """
-        predefined_months = list(MONTH_INT_TO_STR.values())
-        input_months = list(chain.from_iterable(custom_seasons))
-
-        if len(input_months) != len(predefined_months):
-            raise ValueError(
-                "Exactly 12 months were not passed in the list of custom seasons."
-            )
-        if len(input_months) != len(set(input_months)):
-            raise ValueError(
-                "Duplicate month(s) were found in the list of custom seasons."
-            )
-
-        for month in input_months:
-            if month not in predefined_months:
-                raise ValueError(
-                    f"The following month is not supported: '{month}'. "
-                    f"Supported months include: {predefined_months}."
-                )
-
-        c_seasons = {"".join(months): months for months in custom_seasons}
-
-        return c_seasons
+        ds = ds.sel(  # type: ignore
+            **{self._dim: ~((ds.time.dt.month == 2) & (ds.time.dt.day == 29))}
+        )
+        return ds
```

Comment:
This is the new method to handle dropping leap years when calculating daily climatology/departures.